forte/pipeline.py

```diff
@@ -1334,6 +1351,15 @@ def evaluate(self) -> Iterator[Tuple[str, Any]]:
             assert isinstance(p, Evaluator)
             yield p.name, p.get_result()
 
+    def get_component(self, ref_name) -> Evaluator[Any]:
+        """
+        Call the evaluator in the pipeline by the reference name to get a component.
+
+        """
+        p = self.components[self.ref_names.index(ref_name)]
```

Comment:
Now `ref_name` is a list and we need to go through a linear search. How about we use a dictionary so we can get this easier?